reactivesocket-tck-drivers/src/main/java/io/reactivesocket/tckdrivers/common/TestSubscriber.java

```diff
@@ -0,0 +1,1113 @@
+        return lastThread;
+    }
+
+    /**
+     * Returns a shared list of received onNext values.
+     *
+     * @return a list of received onNext values
+     */
+    public final List<Tuple<String, String>> values() {
+        return values;
+    }
+
+    /**
+     * Returns a shared list of received onError exceptions.
+     *
+     * @return a list of received events onError exceptions
+     */
+    public final List<Throwable> errors() {
+        return errors;
+    }
+
+    /**
+     * Returns the number of times onComplete was called.
+     *
+     * @return the number of times onComplete was called
+     */
+    public final long completions() {
+        return completions;
+    }
+
+    /**
+     * Returns true if TestSubscriber received any onError or onComplete events.
+     *
+     * @return true if TestSubscriber received any onError or onComplete events
+     */
+    public final boolean isTerminated() {
+        return done.getCount() == 0;
+    }
+
+    /**
+     * Returns the number of onNext values received.
+     *
+     * @return the number of onNext values received
+     */
+    public final int valueCount() {
+        return values.size();
+    }
+
+    /**
+     * Returns the number of onError exceptions received.
+     *
+     * @return the number of onError exceptions received
+     */
+    public final int errorCount() {
+        return errors.size();
+    }
+
+    /**
+     * Returns true if this TestSubscriber received a subscription.
+     *
+     * @return true if this TestSubscriber received a subscription
+     */
+    public final boolean hasSubscription() {
+        return subscription.get() != null;
+    }
+
+    /**
+     * Awaits until this TestSubscriber receives an onError or onComplete events.
+     *
+     * @return this
+     * @throws InterruptedException if the current thread is interrupted while waiting
+     * @see #awaitTerminalEvent()
+     */
+    public final TestSubscriber await() throws InterruptedException {
+        if (done.getCount() == 0) {
+            return this;
+        }
+
+        done.await();
+        return this;
+    }
+
+    /**
+     * Awaits the specified amount of time or until this TestSubscriber
+     * receives an onError or onComplete events, whichever happens first.
+     *
+     * @param time the waiting time
+     * @param unit the time unit of the waiting time
+     * @return true if the TestSubscriber terminated, false if timeout happened
+     * @throws InterruptedException if the current thread is interrupted while waiting
+     * @see #awaitTerminalEvent(long, TimeUnit)
+     */
+    public final boolean await(long time, TimeUnit unit) throws InterruptedException {
+        if (done.getCount() == 0) {
+            return true;
+        }
+        return done.await(time, unit);
+    }
+
+    public final boolean awaitAtLeast(long n, long time, TimeUnit unit) throws InterruptedException {
+        numOnNext.await(time, unit);
```

Comment:
I suggest checking the return values for both these await calls, and making sure the test terminates cleanly if there is some failure causing a missing event.